CryptoLib/CryptoLib/SmartCardTokenWrapper.h

```diff
@@ -20,25 +20,29 @@
  *
  */
 
-#ifndef ApduToken_hpp
-#define ApduToken_hpp
 
-#include <stdio.h>
+
+#if __cplusplus
 #import "cdoc/Token.h"
-#import <UIKit/UIKit.h>
-#endif /* ApduToken_hpp */
 
+#include <stdio.h>
+#import "AbstractSmartToken.h"
 
-class CDOC_EXPORT ApduToken: public Token
+#import <UIKit/UIKit.h>
+
+class  SmartCardTokenWrapper: public Token
 {
 public:
-    ApduToken(const std::string &password,  UIViewController *controller);
-    ApduToken();
+    SmartCardTokenWrapper(const std::string &password,  AbstractSmartToken *smartToken);
+    SmartCardTokenWrapper();
     virtual std::vector<uchar> cert() const override;
     virtual std::vector<uchar> decrypt(const std::vector<uchar> &data) const override;
     virtual std::vector<uchar> derive(const std::vector<uchar> &publicKey) const override;
+    
 private:
-    DISABLE_COPY(ApduToken);
     class Private;
     Private *d;
```

Comment:
Type name `Private` doesn't tell anything about it's purpose. 
One-letter member variable names are extremely bad practice.